hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSink.java

```diff
@@ -193,6 +193,7 @@ public void replicateEntries(List<WALEntry> entries, final CellScanner cells,
             for (int i = 0; i < count; i++) {
               // Throw index out of bounds if our cell count is off
               if (!cells.advance()) {
+                this.metrics.incrementFailedBatches();
```

Comment:
Looks like you're catching Exception below, and incrementing there as well. Will this result in double-incrementing for these ArrayIndexOutOfBoundExceptions?

Reply:
The catch below actually should remain an IOException. (seems like my cherry-pick from my working branch didnt catch this change.)

Reply:
Running the mvn verify command locally results in the following error: 

```
➜  2022-04-25 14:54:13Z  hbase git:(HBASE-26581_master) mvn clean verify -pl hbase-server -am -Dhadoop.profile=3.0 -Dtest=org.apache.hadoop.hbase.replication.regionserver.TestReplicationSink 
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs-client:jar is missing. @ line 91, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs:jar is missing. @ line 95, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-jobclient:test-jar is missing. @ line 199, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs:test-jar is missing. @ line 223, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 229, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 234, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 458, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-core:jar is missing. @ line 102, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-common:jar is missing. @ line 112, column 17
[ERROR] 'dependencies.dependency.version' for javax.activation:javax.activation-api:jar is missing. @ line 120, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs-client:jar is missing. @ line 147, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 168, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 242, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 313, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs:test-jar is missing. @ line 73, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-app:test-jar is missing. @ line 79, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-jobclient:test-jar is missing. @ line 85, column 17
[ERROR] 'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 109, column 17
 @ 
[ERROR] The build could not read 11 projects -> [Help 1]
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-balancer:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-balancer/pom.xml) has 2 errors
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs-client:jar is missing. @ line 91, column 17
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs:jar is missing. @ line 95, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-mapreduce:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-mapreduce/pom.xml) has 3 errors
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-jobclient:test-jar is missing. @ line 199, column 17
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs:test-jar is missing. @ line 223, column 17
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 229, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-http:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-http/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 234, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-server:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-server/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 458, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-hadoop-compat:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-hadoop-compat/pom.xml) has 3 errors
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-core:jar is missing. @ line 102, column 17
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-common:jar is missing. @ line 112, column 17
[ERROR]     'dependencies.dependency.version' for javax.activation:javax.activation-api:jar is missing. @ line 120, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-procedure:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-procedure/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs-client:jar is missing. @ line 147, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-endpoint:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-endpoint/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 168, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-examples:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-examples/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 242, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-rest:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-rest/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 313, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-shaded-testing-util:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-shaded/hbase-shaded-testing-util/pom.xml) has 3 errors
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-hdfs:test-jar is missing. @ line 73, column 17
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-app:test-jar is missing. @ line 79, column 17
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-mapreduce-client-jobclient:test-jar is missing. @ line 85, column 17
[ERROR]   
[ERROR]   The project org.apache.hbase:hbase-asyncfs:3.0.0-alpha-3-SNAPSHOT (/Users/baugenreich/src/hbase/hbase-asyncfs/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.apache.hadoop:hadoop-minikdc:jar is missing. @ line 109, column 17
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException

```